Sources/GRPC/GRPCServerRequestRoutingHandler.swift

```diff
@@ -216,25 +216,56 @@ extension GRPCServerRequestRoutingHandler: ChannelInboundHandler, RemovableChann
     }
   }
 
+  /// A call URI split into components.
+  struct CallPath {
+    /// The name of the service to call.
+    var service: String.UTF8View.SubSequence
+    /// The name of the method to call.
+    var method: String.UTF8View.SubSequence
+
+    /// Charater used to split the path into components.
+    private let pathSplitDelimiter = UInt8(ascii: "/")
+
+    /// Split a path into service and method.
+    /// Split is done in UTF8 as this turns out to be approximately 10x faster than a simple split.
+    /// URI format: "/package.Servicename/MethodName"
+    init?(requestURI: String) {
+      var utf8View = requestURI.utf8[...]
+      // Check and remove the split character at the beginning.
+      guard utf8View.trimPrefix(to: self.pathSplitDelimiter)?.isEmpty == true else {
```

Comment:
This is the least ugly, I think.

```suggestion
      guard let prefix = utf8View.trimPrefix(to: self.pathSplitDelimiter), prefix.isEmpty else {
```

Reply:
Slightly less dark magic but adds another name `prefix` into scope below - no strong feelings really.

Reply:
Agreed that all ways are a bit ugly but the extra `prefix` in the scope below is better than the `== true`, I think.